include/mbedtls/aes.h

```diff
@@ -212,6 +281,72 @@ int mbedtls_aes_crypt_cbc( mbedtls_aes_context *ctx,
                     unsigned char *output );
 #endif /* MBEDTLS_CIPHER_MODE_CBC */
 
+#if defined(MBEDTLS_CIPHER_MODE_XTS)
+/**
+ * \brief      This function performs an AES-XEX encryption or decryption
+ *             operation for an entire XEX data unit.
+ *
+ *             Length must be a multiple of the AES block size (16 bytes).
+ *
+ * \param ctx          The AES XTS context to use for AES XEX operations.
+ * \param mode         The AES operation: #MBEDTLS_AES_ENCRYPT or
+ *                     #MBEDTLS_AES_DECRYPT.
+ * \param length       The of the input data in bytes. The length must be a
+ *                     multiple of 16 bytes.
+ * \param iv           The initialization vector (not updated after use).
+ * \param input        The buffer holding the input data.
+ * \param output       The buffer holding the output data.
+ *
+ * \return             \c 0 on success.
+ * \return             #MBEDTLS_ERR_AES_INVALID_INPUT_LENGTH if length smaller
+ *                     than an AES block in size (16 bytes) or if the length is
+ *                     larger than 2**20 blocks.
+ */
+int mbedtls_aes_crypt_xex( mbedtls_aes_xts_context *ctx,
+                           int mode,
+                           size_t length,
+                           const unsigned char iv[16],
+                           const unsigned char *input,
+                           unsigned char *output );
+
+/**
+ * \brief      This function performs an AES-XTS encryption or decryption
+ *             operation for an entire XTS data unit.
+ *
+ *             AES-XTS encrypts or decrypts blocks based on their location as
+ *             defined by a data unit number. These must be provided by \p
+ *             data_unit.
+ *
+ *             NIST SP 800-38E limits the maximum size of a data unit to 2**20
+ *             AES blocks. If the data unit is larger than this, this function
+ *             returns #MBEDTLS_ERR_AES_INVALID_INPUT_LENGTH.
+ *
+ * \param ctx          The AES XTS context to use for AES XTS operations.
+ * \param mode         The AES operation: #MBEDTLS_AES_ENCRYPT or
+ *                     #MBEDTLS_AES_DECRYPT.
+ * \param length       The length of both an entire data unit and the input
+ *                     data in bytes. The length must be at least 16 bytes. The
+ *                     length does not need to be a multiple of 16 bytes.
+ * \param data_unit    The data unit (commonly a block device sector) address
```

Comment:
Updated the description.

`data_unit` seems suitable shorthand for the data unit address, just as `sector` would be for the sector address. This is not nominative or anything, just own rationale: I think of CHS as cylinder-head-sector, not cylinder-address-head-address-sector-address, even though the latter is more accurate.

Reply:
When I read the parameter name `data_unit`, I expected it to be a buffer, something of type `uint8_t*` with an associated `size_t data_unit_len`. The original description reinforced this because it started with “the data unit [blah blah]”. It's a lot better now that the description starts with “the address”, but I still find the parameter name misleading, and I'd find `sector` misleading as well if the context was where one would expect to find the sector content. It's different in the context “cylinder-head-sector” because there the words are units, like “degrees-minutes-seconds”.